src/Setup/UpdateRunner.cpp

```diff
@@ -259,8 +259,12 @@ int CUpdateRunner::ExtractUpdaterAndRun(wchar_t* lpCommandLine, bool useFallback
 		lpCommandLine = L"";
 	}
 
-	wchar_t cmd[MAX_PATH];
-	swprintf_s(cmd, L"\"%s\" --install . %s", updateExePath, lpCommandLine);
+	// append the full path of this installer executable to the next stage of the process
+	wchar_t setupExePath[MAX_PATH];
+	GetModuleFileName(NULL, setupExePath, MAX_PATH);
+
+	wchar_t cmd[MAX_PATH * 2];
```

Comment:
Does it make sense to do this? Since we're appending a string which could be up to `MAX_PATH`, I thought it appropriate to add `MAX_PATH` on to this variables previous length, which was also `MAX_PATH`.

Reply:
it's not a concern memory wise. UNC paths can go beyond MAX_PATH btw.